File: SDKs/amazonaws.com/lightsail/2016-11-28/java/lib/src/main/java/openapisdk/models/shared/InstancePortInfo.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * InstancePortInfo
 * Describes information about ports for an Amazon Lightsail instance.
**/
public class InstancePortInfo {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("accessDirection")
    public AccessDirectionEnum accessDirection;
    public InstancePortInfo withAccessDirection(AccessDirectionEnum accessDirection) {
        this.accessDirection = accessDirection;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("accessFrom")
    public String accessFrom;
    public InstancePortInfo withAccessFrom(String accessFrom) {
        this.accessFrom = accessFrom;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("accessType")
    public PortAccessTypeEnum accessType;
    public InstancePortInfo withAccessType(PortAccessTypeEnum accessType) {
        this.accessType = accessType;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("cidrListAliases")
    public String[] cidrListAliases;
    public InstancePortInfo withCidrListAliases(String[] cidrListAliases) {
        this.cidrListAliases = cidrListAliases;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("cidrs")
    public String[] cidrs;
    public InstancePortInfo withCidrs(String[] cidrs) {
        this.cidrs = cidrs;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("commonName")
    public String commonName;
    public InstancePortInfo withCommonName(String commonName) {
        this.commonName = commonName;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("fromPort")
    public Long fromPort;
    public InstancePortInfo withFromPort(Long fromPort) {
        this.fromPort = fromPort;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("ipv6Cidrs")
    public String[] ipv6Cidrs;
    public InstancePortInfo withIpv6Cidrs(String[] ipv6Cidrs) {
        this.ipv6Cidrs = ipv6Cidrs;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("protocol")
    public NetworkProtocolEnum protocol;
    public InstancePortInfo withProtocol(NetworkProtocolEnum protocol) {
        this.protocol = protocol;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("toPort")
    public Long toPort;
    public InstancePortInfo withToPort(Long toPort) {
        this.toPort = toPort;
        return this;
    }
}

File: SDKs/amazonaws.com/lightsail/2016-11-28/python/src/sdk/models/shared/instanceportinfo.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import accessdirection_enum as shared_accessdirection_enum
from ..shared import portaccesstype_enum as shared_portaccesstype_enum
from ..shared import networkprotocol_enum as shared_networkprotocol_enum


@dataclass_json
@dataclasses.dataclass
class InstancePortInfo:
    r"""InstancePortInfo
    Describes information about ports for an Amazon Lightsail instance.
    """
    
    access_direction: Optional[shared_accessdirection_enum.AccessDirectionEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('accessDirection') }})
    access_from: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('accessFrom') }})
    access_type: Optional[shared_portaccesstype_enum.PortAccessTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('accessType') }})
    cidr_list_aliases: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cidrListAliases') }})
    cidrs: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cidrs') }})
    common_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('commonName') }})
    from_port: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('fromPort') }})
    ipv6_cidrs: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ipv6Cidrs') }})
    protocol: Optional[shared_networkprotocol_enum.NetworkProtocolEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('protocol') }})
    to_port: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('toPort') }})
    
